File: typescript/ex3_2_2.ts
class Chooser<T> {
    v1: T;
    v2: T;
      
    constructor(v1: T, v2: T) {
        this.v1 = v1;
        this.v2 = v2;
    }

    choose(): T { return true ? this.v1 : this.v2; }
}

let c: Chooser<number> = new Chooser<number>(1, 2);
let n: number = c.choose();

File: rust/ex3_2_2.rs
struct Chooser<T> { v1: T, v2: T }

impl<T> Chooser<T> {
    fn choose(self) -> T {
        if true { self.v1 } else { self.v2 }
    }
}

fn main() {
    let c: Chooser<i32> = Chooser::<i32> { v1: 1, v2: 2 };
    let n: i32 = c.choose();
}
